File: plugins/ghidra/src/main/java/lief/dwarf/editor/Variable.java
/* Copyright 2022 - 2026 R. Thomas
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package lief.dwarf.editor;

public class Variable extends lief.Base {
    @Override
    protected native void destroy();

    private Variable(long impl) {
        super(impl);
    }

    public native Variable setAddr(long addr);

    public native Variable setStackOffset(long offset);

    public native Variable setExternal();

    public native Variable setType(Type ty);

    public native Variable addDescription(String desc);
};

File: api/rust/crates/lief-ffi/src/dwarf/editor/variable.rs
#[cxx::bridge]
pub mod ffi {
    unsafe extern "C++" {
        include!("LIEF/rust/DWARF/editor/Variable.hpp");

        type DWARF_editor_Type = crate::dwarf::editor::type_::ffi::DWARF_editor_Type;

        type DWARF_editor_Variable;

        fn set_external(self: Pin<&mut DWARF_editor_Variable>);
        fn set_addr(self: Pin<&mut DWARF_editor_Variable>, addr: u64);
        fn set_stack_offset(self: Pin<&mut DWARF_editor_Variable>, addr: u64);
        fn set_type(self: Pin<&mut DWARF_editor_Variable>, ty: &DWARF_editor_Type);
        fn add_description(self: Pin<&mut DWARF_editor_Variable>, desc: &CxxString);
    }

    impl UniquePtr<DWARF_editor_Variable> {}
}
